src/core/filesystem.cc

```diff
@@ -1619,6 +1678,25 @@ GetFileSystem(const std::string& path, FileSystem** file_system)
   return Status::Success;
 }
 
+Status
+GetFileSystemByType(FileSystemType type, FileSystem** file_system)
```

Comment:
I agree that the one that takes a path should internally invoke GetFileSystem(type, ...), but it is blocked because the constructor of some remote filesystems requires a path to be passed. That is why I asked if there are tests for multiple remote filesystems earlier.